org.eclipse.lsp4j.jsonrpc/src/test/java/org/eclipse/lsp4j/jsonrpc/test/TracingMessageConsumerTest.java

```diff
@@ -299,6 +299,72 @@ public void testSendingNotificationWithCustomAdapter() {
 
 		assertEquals(expectedTrace, actualTrace);
 	}
+
+	@Test
+	public void testUnmatchedResponseWhenOutgoingIsNotStreamConsumer() {
+		// Simulate an outgoing transport that is not StreamMessageConsumer (e.g., websocket)
+		var outgoingTrace = new StringWriter();
+		var outgoingPrintWriter = new PrintWriter(outgoingTrace);
+
+		var incomingTrace = new StringWriter();
+		var incomingPrintWriter = new PrintWriter(incomingTrace);
+
+		var sentRequests = new HashMap<String, RequestMetadata>();
+		var receivedRequests = new HashMap<String, RequestMetadata>();
+
+		// Wrap a non-stream consumer so TracingMessageConsumer does not treat it as "sending"
```

Comment:
This comment looks a bit misleading, as it describes the old implementation rather than the new one.